### DISCUSSION TITLE
Trouble adding Argentina geojson to country map legacy plugin

### DISCUSSION THREAD

**Discussion by aguspina** [apache/superset#18268] [Q&A / Help]
I connected superset with a local modify version of the superset-ui plugin legacy-country map but I got this viz:

<img width="783" alt="Screen Shot 2021-02-06 at 16 34 26" src="https://user-images.githubusercontent.com/4105740/107128049-6cd33c80-6899-11eb-8d2e-eda969ef041f.png">

I'm the the right geojson generated from the file downloaded from https://www.diva-gis.org/gdata with de ISO column updated to follow de ISO 3166-2 standard.

If someone help me solve this, I will make the corresponding PR!

**Comment by aguspina** [comment]
Here is the geojson file https://gist.github.com/aguspina/570fe8c52bb9628f38618ad9b037f4e7

**Comment by junlincc** [comment]
@aguspina can you provide detailed steps to reproduce this issue? thanks!

**Comment by aguspina** [comment]
Of course :)

I added this line to the countries.js

```python
import argentina from 'file-loader!./countries/argentina.geojson';
```

The argentina.geojson file into the countries folder and added argentina to the choices in the controlPanel.ts file
```typescript
label: t('Country Name'),
              default: 'France',
              choices: [
                'Argentina',
```

And I had to change the package json in superset-frontend to ponint to my local fork of superset-ui plugins

**Comment by junlincc** [comment]
@aguspina thanks for the details.  @mayurnewase do you mind taking a look this issue? 🙏

**Comment by mayurnewase** [comment]
I think the file is in a different format, I found this pr fixes Bulgaria map, which suffers from a similar issue.

**Comment by djok** [comment]
Please test this geoJson https://gist.github.com/djok/b53cbbb4f76ecaa8499fdafbc1cac040

I used mapshaper.org to simplify and fix overlapping

**Comment by aguspina** [comment]
Hi @djok thanks for your help. I think the geojson you shared to me is malformed, it is duplicated and truncated.

**Comment by bdiouih** [comment]
> Please test this geoJson https://gist.github.com/djok/b53cbbb4f76ecaa8499fdafbc1cac040
> 
> I used mapshaper.org to simplify and fix overlapping

I have exactly the similar issue with Afghanistan geojson file.
How can this be solved exactly  :  how can it be simplified and overlapping fixed?

**Comment by kakoni** [comment]
I was testing country map with some addiotional geojsons. I found out that in somecases center point was calculated wrong. There is an discussion about it at https://github.com/d3/d3/issues/1883

**Comment by rukshn** [comment]
Anyone else solved this issue? I'm facing a similar problem
